### ISSUE TITLE
SIGABRT / corrupted double-linked list

### ISSUE THREAD

**Issue by jcaesar** [stedolan/jq#2178]
**Describe the bug**
jq crashes with the input / invocation below.

**To Reproduce**
```bash
echo '[ {
    "committer_date" : "2020-08-26T07:44:47.000Z",
    "usage_queued_at" : "2020-08-26T07:46:31.836Z",
    "parallel" : 3,
    "branch" : "ddev-ghesync/pv2",
    "author_date" : "2020-08-26T07:44:47.000Z",
    "why" : "github",
    "vcs_tag" : null,
    "build_num" : 5266,
    "status" : "success",
    "committer_name" : "GitHub Enterprise",
    "subject" : "Merge pull request #629 from ms/o",
    "dont_build" : null,
    "lifecycle" : "finished",
    "fleet" : "picard",
    "stop_time" : "2020-08-26T07:55:06.372Z",
    "build_time_millis" : 511972,
    "start_time" : "2020-08-26T07:46:34.400Z",
    "platform" : "2.0",
    "outcome" : "success",
    "queued_at" : "2020-08-26T07:46:31.871Z"
  } ]' | jq -ja '. | @json + "\u0000"'
```

**Expected behavior**
Guess!

**Environment (please complete the following information):**
 - OS and Version: Arch Linux - updated yesterday, CentOS 7
 - `$ jq --version → jq-1.6`

**Comment by vintnes** [comment]
I can't comment on the ascii/unicode translation, but It looks like you're trying to generate null-delimited output.

Why not generate/parse newline-delimited output with `--compact-output`?

**Comment by jcaesar** [comment]
I opted for dropping the ascii translation.
```bash
... \
| jq -cMj '... | @json + "\u0000"' \
| while IFS= read -rd $'\0' build; do
     ...
done
```
I was hunting for a problem where the escapes around a quotation mark in a string were lost. `read` without its `-r` was at fault. So no, there is not really a reason for me to generate 0 terminated output anymore.

Nevertheless, most people probably wouldn't want jq to crash on funky input, so I reported this issue. Maybe there are some less far-fetched cases.

**Comment by vintnes** [comment]
Maybe I'm missing something obvious, but what is the purpose of adding/splitting nulls?
AFAIK `jq -c` generates valid JSONL:

```
jq -cn '{foo: ("bar", "baz", "qux")}' |
  while IFS= read -r json; do
    jq . <<< "$json" || echo failed
  done
```

**Comment by jcaesar** [comment]
Maybe I haven't said it clearly enough: There is no point in what I did, and the purpose was based on wrong assumptions. Does the purpose matter for this issue?

**Comment by jcaesar** [comment]
In fact, I had forgotten to minify the filter fully. It is sufficient to use only `@json` to cause a crash.

**Comment by pkoppstein** [comment]
For the record, I was not able to reproduce the reported problem on a Mac using jq 1.5, jq 1.6, and a recent "master" version. (The -j option was only introduced in jq 1.5.)

And by the way, that initial `.` in the jq pipeline can be omitted.

And, yes, `jq -c` produces valid JSONL so long as some other option (notably -j or -r) does not subvert it.

**Comment by wtlangford** [comment]
I can't reproduce this, and without the crash output, I can't find what caused the issue. If you're still able to reproduce this, please reopen the issue and share the crash output.

**Comment by jcaesar** [comment]
I bisected because I was worried that this was maybe a problem that got masked by optimizations playing  out a different way due to unrelated changes. It seems that a proper fix was introduced in a1f1231a73c221155d539a281181ef37f874869d. All good.
